src/async-wrap.cc

```diff
@@ -212,6 +212,8 @@ void AsyncWrap::DestroyIdsCb(uv_idle_t* handle) {
       FatalException(env->isolate(), try_catch);
     }
   }
+
+  env->destroy_ids_list()->clear();
```

Comment:
Can I suggest you swap before iteration?  I.e.:
```c++
std::vector<int64_t> destroy_ids_list;
destroy_ids_list.swap(*env->destroy_ids_list());
for (auto current_id : destroy_ids_list) {
  // ...
}
```
That way the list is both cleared and immune to concurrent modification by the callback.